sdk/src/wallet/operations/announce_candidacy.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    client::secret::SecretManage,
+    types::block::{
+        output::AccountId,
+        payload::{CandidacyAnnouncementPayload, Payload},
+        BlockId,
+    },
+    wallet::Wallet,
+};
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    crate::wallet::Error: From<S::Error>,
+    crate::client::Error: From<S::Error>,
+{
+    /// Announce a staking account's candidacy for the staking period.
+    pub async fn announce_candidacy(
+        &self,
+        account_id: impl Into<Option<AccountId>> + Send,
+    ) -> crate::wallet::Result<BlockId> {
+        self.submit_basic_block(Payload::CandidacyAnnouncement(CandidacyAnnouncementPayload), account_id)
```

Comment:
Should `begin_staking()` allot extra Mana for this block or how do we want to do this?

Reply:
🤷‍♀️ I guess that could make sense but at the same time you need to make one of these every epoch right?

Reply:
Yes, but staking feature you also need to update every epoch

Reply:
you do? are you sure? it seems like you can keep it for many epochs

Reply:
True, misread something, yeah then I guess people should just allot themselves when needed